pkg/operator/controller/dns/controller.go

```diff
@@ -661,26 +660,30 @@ func getIbmDNSProvider(dnsConfig *configv1.DNS, creds *corev1.Secret, instanceID
 	}
 
 	providerCfg := ibm.Config{
-		APIKey:     string(creds.Data["ibmcloud_api_key"]),
-		InstanceID: instanceID,
-		Zones:      zones,
-		UserAgent:  userAgent,
+		APIKey:    string(creds.Data["ibmcloud_api_key"]),
+		Zones:     zones,
+		UserAgent: userAgent,
 	}
 
 	if isPublic {
+		providerCfg.InstanceID = instanceCRN
 		provider, err := ibmpublicdns.NewProvider(providerCfg)
 		if err != nil {
 			return nil, fmt.Errorf("failed to initialize IBM CIS DNS provider: %w", err)
 		}
 		log.Info("successfully initialized IBM CIS DNS provider")
 		return provider, nil
 	} else {
+		matches := ibm.IBMResourceCRNRegexp.FindStringSubmatch(instanceCRN)
```

Comment:
It will be great if you can add some comment about the regex format we are talking about.

Reply:
We have it over [here](https://github.com/Karthik-K-N/cluster-ingress-operator/blob/10966557ef8592858fe068065e526a87c37315c5/pkg/dns/ibm/common.go#L14-L16).